src/ristretto.rs

```diff
@@ -245,21 +245,25 @@ impl CompressedRistretto {
         // Step 2.  The rest.  (XXX write comments)
         let one = FieldElement::one();
         let ss = s.square();
-        let yden = &one + &ss; // 1 - a*s^2
-        let ynum = &one - &ss; // 1 + a*s^2
-        let yden_sqr = yden.square();
-        let xden_sqr = &(&(-&constants::EDWARDS_D) * &ynum.square()) - &yden_sqr;
+        let u1 = &one + &ss;      // 1 + as²
```

Comment:
Since `a = -1`, `1 + as^2` is `1 - s^2`, not `1 + s^2`, so I think this should be `u2` (and the variable named `u2` is actually `u1`.

Reply:
I think that rather than swapping the roles of `u1` and `u2` in the formulas here: https://github.com/Ristretto/website/pull/24/files, the `u1` and `u2` have been swapped here, because the code is using the fact that `a = -1` to compute `1 \pm as^2` as `1 \mp s^2` (i.e., avoiding a multiplication by `-1`).

Reply:
I've now flipped the comments back and added a note that `a=-1`.